erroranalysis/erroranalysis/_internal/surrogate_error_tree.py

```diff
@@ -739,6 +741,36 @@ def node_to_dict(df, tree, nodeid, categories, json,
     return json, df
 
 
+def filter_by_threshold(df, p_node_name_val, parent_threshold, side):
+    """Filters the DataFrame by the threshold of the parent node.
+
+    :param df: The DataFrame to filter.
+    :type df: pandas.DataFrame
+    :param p_node_name_val: The name of the parent node.
+    :type p_node_name_val: str
+    :param parent_threshold: The threshold of the parent node.
+    :type parent_threshold: float
+    :param side: The side of the parent node.
```

Comment:
Return type of pandas dataframe missing.

Reply:
good catch, done!